src/features/basic_spa/instrument/index.js

```diff
@@ -0,0 +1,60 @@
+import { originals } from '../../../common/config/config'
+import { isBrowserScope } from '../../../common/constants/runtime'
+import { handle } from '../../../common/event-emitter/handle'
+import { windowAddEventListener } from '../../../common/event-listener/event-listener-opts'
+import { now } from '../../../common/timing/now'
+import { debounce } from '../../../common/util/invoke'
+import { wrapEvents, wrapHistory } from '../../../common/wrap'
+import { InstrumentBase } from '../../utils/instrument-base'
+import { CATEGORY, FEATURE_NAME, INTERACTION_EVENTS } from '../constants'
+
+export class Instrument extends InstrumentBase {
+  static featureName = FEATURE_NAME
+  constructor (agentIdentifier, aggregator, auto = true) {
+    super(agentIdentifier, aggregator, FEATURE_NAME, auto)
+    if (!isBrowserScope) return // SPA not supported outside web env
+    try {
+      this.removeOnAbort = new AbortController()
+    } catch (e) {}
+
+    const historyEE = wrapHistory(this.ee)
+    const eventsEE = wrapEvents(this.ee)
+
+    const trackURLChange = (args) => handle('newURL', [now(), '' + window.location, args?.type], undefined, this.featureName, this.ee)
+
+    historyEE.on('pushState-end', trackURLChange)
+    historyEE.on('replaceState-end', trackURLChange)
+
+    windowAddEventListener('hashchange', trackURLChange, true, this.removeOnAbort?.signal)
+    windowAddEventListener('load', trackURLChange, true, this.removeOnAbort?.signal)
+    windowAddEventListener('popstate', trackURLChange, true, this.removeOnAbort?.signal)
```

Comment:
Do we need both the history wrappers on lined 25-26 and this event listener?

Reply:
These are ports over from the original spa feature, needs eval as to what is actually needed there